evaluator/src/main/resources/rules/osadl.rules.kts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2022 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+// Use this rule like:
+//
+// $ ort evaluate -i scanner/src/funTest/assets/dummy-expected-output-for-analyzer-result.yml --rules-resource /rules/osadl.rules.kts
+
+import org.ossreviewtoolkit.evaluator.osadl.Compatibility
+import org.ossreviewtoolkit.evaluator.osadl.matrix
+
+val ruleSet = ruleSet(ortResult, licenseInfoResolver) {
+    val licenseView = LicenseView.CONCLUDED_OR_DECLARED_AND_DETECTED
+
+    // Define a rule that is executed for each dependency.
+    dependencyRule("OSADL compliance matrix") {
```

Comment:
We always have used rules names as all caps with underscores (kind of enums) see https://github.com/oss-review-toolkit/ort/blob/main/examples/evaluator-rules/src/main/resources/example.rules.kts. Would like us to be consistent across the ORT project

Reply:
THB the convention to use as all caps with underscores here never made sense to me. IMO this should be more of a speaking value, similar to the rule violation messages themselves, so it also reads nicely which rule actually failed. I'd rather change the other existing rules to have speaking names, too.

Reply:
> Update to ORT's documentation is missing so this would basically be a hidden ORT feature. Is this intentional?

This PR is pending already for way too long, and I don't want to hold it back further due to missing docs. I'll add them later.

Reply:
Using all caps (with underscores) has the advantage of being able to use the string as-is as input to a (helper-CLI command). Personally, I also like all caps more than free text from, because the textual representation is the "message". That message does not require further text as it stands on its own. So, in my view - the 'name' needs to be as easily / quickly recognizable as possible by the human brain. In my case this works better with all caps. So I vote for it.

Reply:
I've changed it to be able to move forward.